src/chem/viirs_aod2ioda.py

```diff
@@ -9,189 +10,237 @@
+                        uncertainty[i] = uncertainty1[i]
+                    else:   # case of dark land high quality
+                        uncertainty[i] = uncertainty2[i]
+                    errs[i] = uncertainty[i]
+
+            #  Write out data
+
+            #  Values
+
+            if first:
+                self.outdata[self.varDict[iodavar]['valKey']] = vals
+                self.outdata[self.varDict[iodavar]['errKey']] = errs
+                self.outdata[self.varDict[iodavar]['qcKey']] = qcall
+            else:
+                self.outdata[self.varDict[iodavar]['valKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['valKey']], vals))
+                self.outdata[self.varDict[iodavar]['errKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['errKey']], errs))
+                self.outdata[self.varDict[iodavar]['qcKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['qcKey']], qcall))
+
+            #  Add Meta data
+
+            if first:
+                self.outdata[('latitude', 'MetaData')] = lats
+                self.outdata[('longitude', 'MetaData')] = lons
+                self.outdata[('datetime', 'MetaData')] = obs_time
+            else:
+                self.outdata[('latitude', 'MetaData')] = np.concatenate((self.outdata[('latitude', 'MetaData')], lats))
+                self.outdata[('longitude', 'MetaData')] = np.concatenate((self.outdata[('longitude', 'MetaData')], lons))
+                self.outdata[('datetime', 'MetaData')] = np.concatenate((self.outdata[('datetime', 'MetaData')], obs_time))
+
+            first = False
+
+        DimDict['nlocs'] = len(self.outdata[('latitude', 'MetaData')])
+        AttrData['nlocs'] = np.int32(DimDict['nlocs'])
+
 
 def main():
 
+    # get command line arguments
+    # Usage: python blah.py -i /path/to/obs/2021060801.nc /path/to/obs/2021060802.nc ... -t Analysis_time /path/to/obs/2021060823.nc
+    # -o /path/to/ioda/20210608.nc
+    # where the input obs could be for any desired interval to concatenated together. Analysis time is generally the midpoint of
+    # analysis window.
     parser = argparse.ArgumentParser(
         description=('Read VIIRS aerosol optical depth file(s) and Converter'
                      ' of native NetCDF format for observations of optical'
-                     ' depth from VIIRS AOD550 to IODA netCDF format.')
+                     ' depth from VIIRS AOD550 to IODA-V2 netCDF format.')
     )
-    parser.add_argument('-i', '--input',
-                        help="name of viirs aod input file(s)",
-                        type=str, required=True)
-    parser.add_argument('-o', '--output',
-                        help="name of ioda output file",
-                        type=str, required=True)
+    parser.add_argument(
+        '-i', '--input',
+        help="path of viirs aod input file(s)",
+        type=str, nargs='+', required=True)
```

Comment:
@andytangborn so I'm confused based on our discussion. What does then `args.input` look like if you do `print(args.input)`?  Should that not be just a list of filenames that then the converter can act on? Like look here: https://github.com/JCSDA-internal/ioda-converters/blob/7a29f9fac0b0e77fdf0f8b5beeb571b76f1789fd/src/chem/tropomi_no2_nc2ioda.py#L195 and then I can run it in a shell script similar to this: `$pybin -i $ObsDir/S5P_OFFL_L2__NO2____$NowDate* -o $OutRoot/$NowTime/tropomi_no2_${NowTime}.nc`